cfme/configure/access_control.py

```diff
@@ -168,149 +238,298 @@ def description(self):
-             action=form_buttons.save)
-        flash.assert_success_message(
-            'Group "{}" was saved'.format(updates.get('description', self.description)))
+        view = navigate_to(self, 'Edit')
+        changed = view.fill({
+            'description_txt': updates.get('description'),
+            'role_select': updates.get('role'),
+            'group_tenant': updates.get('tenant')
+        })
+        if changed:
+            view.save_button.click()
+            flash_message = 'Group "{}" was saved'.format(
+                updates.get('description', self.description))
+        else:
+            view.cancel_button.click()
+            flash_message = 'Edit of Group was cancelled by the user'
+        view = self.create_view(DetailsGroupView, override=updates)
+        view.flash.assert_message(flash_message)
+        assert view.is_displayed
 
     def delete(self):
-        navigate_to(self, 'Details')
-        tb_select('Delete this Group', invokes_alert=True)
-        sel.handle_alert()
-        flash.assert_success_message('EVM Group "{}": Delete successful'.format(self.description))
+        view = navigate_to(self, 'Details')
+        view.configuration.item_select('Delete this Group', handle_alert=True)
+        view = self.create_view(AllGroupView)
+        view.flash.assert_success_message(
+            'EVM Group "{}": Delete successful'.format(self.description))
+        assert view.is_displayed
 
     def edit_tags(self, tag, value):
-        navigate_to(self, 'Details')
-        pol_btn("Edit 'My Company' Tags for this Group", invokes_alert=True)
-        fill(edit_tags_form, {'select_tag': tag,
-                              'select_value': value},
-             action=form_buttons.save)
-        flash.assert_success_message('Tag edits were successfully saved')
+        view = navigate_to(self, 'EditTags')
+        view.fill({'select_tag': tag,
+                   'select_value': value})
+        view.save_button.click()
+        view = self.create_view(DetailsGroupView)
+        view.flash.assert_success_message('Tag edits were successfully saved')
+        assert view.is_displayed
 
     def remove_tag(self, tag, value):
-        navigate_to(self, 'Details')
-        pol_btn("Edit 'My Company' Tags for this Group", invokes_alert=True)
-        row = tag_table.find_row_by_cells({'category': tag, 'assigned_value': value},
-                                          partial_check=True)
-        sel.click(row[0])
-        form_buttons.save()
-        flash.assert_success_message('Tag edits were successfully saved')
+        view = navigate_to(self, 'EditTags')
+        row = view.tag_table.row((1, tag), (2, value))
+        row[0].click()
+        view.save_button.click()
+        view = self.create_view(DetailsGroupView)
+        view.flash.assert_success_message('Tag edits were successfully saved')
+        assert view.is_displayed
+
+    def set_group_order(self, items):
```

Comment:
It would be good if these two methods became a settable property `group_order`. That would implicitly by semantics require that whatever comes out of it can be fed in it.